libs/workspace-manager/src/lib/features/filesystem-navigator/services/filesystem-navigator-utils.service.ts

```diff
@@ -0,0 +1,34 @@
+import { inject, Injectable } from '@angular/core';
+import { Directory } from '@angular-cli-gui/shared/data';
+import { map, Observable } from 'rxjs';
+
+import { WorkspaceManagerApiService } from '../../../data-access/workspace-manager-api.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class FilesystemNavigatorUtilsService {
+  private readonly workspaceManagerApiService = inject(
+    WorkspaceManagerApiService
+  );
+
+  extractDirNameFromPath(path: string): string {
+    const pathParts = path.split('/');
+    return pathParts[pathParts.length - 1];
+  }
+
+  getDirectoryFromParentPath(
+    path: string,
+    dirname: string
+  ): Observable<Directory | undefined> {
+    const parentPath = this.getParentPath(path);
+    return this.workspaceManagerApiService
+      .getDirectoriesInPath(parentPath)
+      .pipe(map((directories) => directories.find((d) => d.name === dirname)));
+  }
+
+  getParentPath(path: string): string {
+    const pathParts = path.split('/');
+    return pathParts.slice(0, pathParts.length - 1).join('/');
+  }
+}
```

Comment:
I don't think this service should exist, create a utils directory on the same level as features and data access, move the getParentPath and extractDirNameFromPath to a pure functions in this directory. 

I think this `getDirectoryFromParentPath` should come from the component that uses the filesystem-navigator

Reply:
I tend to agree but this is a very common use-case that will need to be handled every time the user clicks on one of the breadcrumbs (path part), so the implementation will always be the same on every consumer of the component.
it will need to get the directories of the  parent path and find the current directory